File: 2019.7/q2studio-2019.7.0/q2studio/api/formats.py
# ----------------------------------------------------------------------------
# Copyright (c) 2016-2019, QIIME 2 development team.
#
# Distributed under the terms of the Modified BSD License.
#
# The full license is in the file LICENSE, distributed with this software.
# ----------------------------------------------------------------------------

from flask import Blueprint, jsonify
import qiime2.sdk

formats = Blueprint('formats', __name__)

PLUGIN_MANAGER = qiime2.sdk.PluginManager()


@formats.route('/importable', methods=['POST'])
def get_importable_formats():
    ret = list(PLUGIN_MANAGER.importable_formats)
    return jsonify(ret)

File: 2019.7/q2studio-2019.7.0/app/js/actions/formats.js
// ----------------------------------------------------------------------------
// Copyright (c) 2016-2019, QIIME 2 development team.
//
// Distributed under the terms of the Modified BSD License.
//
// The full license is in the file LICENSE, distributed with this software.
// ----------------------------------------------------------------------------

import { fetchAPI } from '../util/auth';

export const importFormats = importableFormatsList => ({
    type: 'IMPORTABLE_FORMATS',
    importableFormatsList
});

export const checkImportableFormats = () => {
    return (dispatch, getState) => {
        const {
            connection: { uri, secretKey }
        } = getState();
        const url = `http://${uri}/api/formats/importable`;
        // TODO: don't hit the server if there is nothing new to ask...
        fetchAPI(secretKey, 'POST', url)
        .then(json => dispatch(importFormats(json)));
    };
};
